### ISSUE TITLE
tf.layers.conv3d throws an error when using 'channels_first' and 'None' size for the input shape

### ISSUE THREAD

**Issue by RedMindZ** [tensorflow/tensorflow#20379]
### System information
- **Have I written custom code**: Yes
- **OS Platform and Distribution**: Windows 10 Home version 1803
- **TensorFlow installed from**: Installed by running "pip install tensorflow-gpu"
- **TensorFlow version (use command below)**: b'v1.8.0-0-g93bc2e2072' 1.8.0
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: CUDA v9.0, cuDNN v7.1
- **GPU model and memory**: GeForce GTX1080, 8GB
- **Exact command to reproduce**: see the *Code To Reproduce The Bug* section

### Bug Description
When calling "tf.layers.conv3d" with "data_format='channels_first'" and the
input shape (None, 1, 3, None, None), tensorflow throws the error:
"TypeError("unsupported operand type(s) for *: 'int' and 'NoneType'",)"

This error appears to be thrown from [this line](https://github.com/tensorflow/tensorflow/blob/f202958ee2d5177a474e3d107fdbf0c83174d099/tensorflow/python/keras/layers/convolutional.py#L205) according to Visual Studio's python debugger.

### Code To Reproduce The Bug
```python
import tensorflow as tf

def WorkingCode():
    InputTensor = tf.placeholder(dtype=tf.float32, shape=(None, 3, None, None, 1))
    OutputTensor = tf.layers.conv3d(inputs=InputTensor, filters=16, kernel_size=(1, 3, 3), data_format='channels_last')

def BuggedCode():
    InputTensor = tf.placeholder(dtype=tf.float32, shape=(None, 1, 3, None, None))
    OutputTensor = tf.layers.conv3d(inputs=InputTensor, filters=16, kernel_size=(1, 3, 3), data_format='channels_first')
```
Calling the `BuggedCode` function will throw the error, while calling `WorkingCode` will work perfectly fine.

### Full Traceback
```
Traceback (most recent call last):
  File ".\TensorflowBugTest.py", line 11, in <module>
    BuggedCode()
  File ".\TensorflowBugTest.py", line 9, in BuggedCode
    OutputTensor = tf.layers.conv3d(inputs=InputTensor, filters=16, kernel_size=(1, 3, 3), data_format='channels_first')
  File "C:\Users\User\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\layers\convolutional.py", line 828, in conv3d
    return layer.apply(inputs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\layers\base.py", line 828, in apply
    return self.__call__(inputs, *args, **kwargs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\layers\base.py", line 717, in __call__
    outputs = self.call(inputs, *args, **kwargs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\layers\convolutional.py", line 187, in call
    outputs_shape[2] * outputs_shape[3],
TypeError: unsupported operand type(s) for *: 'int' and 'NoneType'
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by cy89** [comment]
@fchollet it looks like you might have introduced the code that manipulates outputs_4d in the case where channels_first is True. Would you PTAL?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shiyangc-intusurg** [comment]
I have a pull request for this in:
https://github.com/tensorflow/tensorflow/pull/21610

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 92 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 106 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by payamr** [comment]
This is still a problem, also for 'channels_last", where if I pass a data type with shape (None, 10, None, None, 1), line 1088 in /keras/layers/convolutional.py drops an error:

outputs_4d = array_ops.reshape(outputs, [
            outputs_shape[0], outputs_shape[1] * outputs_shape[2],
            outputs_shape[3], outputs_shape[4]
        ])

TypeError: unsupported operand type(s) for *: 'int' and 'NoneType'

**Comment by berger-stx** [comment]
Might be fixed upstream, but this is still a problem in `1.13.1`. 

```python
import tensorflow.keras as keras
inp = keras.layers.Input([4, None, None, None])
keras.layers.Conv3D(16, 3, data_format='channels_first')(inp)
```

throws 

```
TypeError                                 Traceback (most recent call last)
<ipython-input-9-97dcda7235cf> in <module>
      1 import tensorflow.keras as keras
      2 inp = keras.layers.Input([4, None, None, None])
----> 3 keras.layers.Conv3D(16, 3, data_format='channels_first')(inp)

/usr/local/lib/python3.5/dist-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, inputs, *args, **kwargs)
    552             # In graph mode, failure to build the layer's graph
    553             # implies a user-side bug. We don't catch exceptions.
--> 554             outputs = self.call(inputs, *args, **kwargs)
    555           else:
    556             try:

/usr/local/lib/python3.5/dist-packages/tensorflow/python/keras/layers/convolutional.py in call(self, inputs)
    211           outputs_4d = array_ops.reshape(outputs,
    212                                          [outputs_shape[0], outputs_shape[1],
--> 213                                           outputs_shape[2] * outputs_shape[3],
    214                                           outputs_shape[4]])
    215           outputs_4d = nn.bias_add(outputs_4d, self.bias, data_format='NCHW')

TypeError: unsupported operand type(s) for *: 'NoneType' and 'NoneType'
```

**Comment by ymodak** [comment]
This is fixed with tf-nightly version '1.15.0-dev20190726'. Thanks!
